### PR TITLE
Mathew feat implement exploration analysis

### PR DESCRIPTION
**Pull request by matemq**

### CODE REVIEW

File: conf/base/catalog_exploration.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+agregated_dataset:
+  type: MemoryDataset
```
Comment (matemq) [inline conf/base/catalog_exploration.yml]: Seems like unrelated change, please separate it into another commit

File: workshop/pipelines/exploration/pipeline.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+def create_pipeline(*args: P.args, **kwargs: P.kwargs) -> Pipeline:
+    aggregated_dataset = "agregated_dataset"
+    return pipeline(
+        pipe=[
+            node(
+                func=load_and_aggregate_data,
+                inputs=None,
+                outputs=aggregated_dataset,
+            ),
+        # node(
```
Comment (matemq) [inline workshop/pipelines/exploration/pipeline.py:19]: this comment seems unnecessary, please remove it

File: workshop/pipelines/exploration/nodes.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import polars as pl
```
Comment (matemq) [inline workshop/pipelines/exploration/nodes.py:1]: this is not working, I'm getting:

```
ModuleNotFoundError: No module named 'polars'
```
when trying to run locally

looks like you should rebase on master

File: workshop/pipelines/exploration/pipeline.py (hunk 2 above)
Comment (HonzaSinkora) [reply workshop/pipelines/exploration/pipeline.py:19]: issue: use conventional comments

Comment (matemq) [reply workshop/pipelines/exploration/pipeline.py]: later

File: workshop/extras/datasets/polars.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+from pathlib import Path, PurePosixPath
+from typing import Any
+
+import fsspec  # type: ignore[import-untyped]
+import polars as pl
+from kedro.io.core import AbstractDataset, get_filepath_str, get_protocol_and_path
+
+
+class PolarsDataSet(AbstractDataset[pl.DataFrame, pl.DataFrame]):
```
Comment (matemq) [inline workshop/extras/datasets/polars.py:9]: This is great! I would like to use in my branch - could you please open new PR with this change so I do not have to wait till merge of this PR

File: conf/base/catalog_exploration.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+agregated_dataset:
```
Comment (matemq) [inline conf/base/catalog_exploration.yml:1]: Looks like there is a lot of unnecessary commit - could you please rewrite the whole branch history and make it little bit more logical. Thank you <3

File: workshop/pipelines/exploration/nodes.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+
+
+def load_and_aggregate_data() -> pl.DataFrame:
+    df = pl.DataFrame({
+        "name": ["Jakub", "Jirka", "Jolana"],
+        "age": ["22", "23", "24"],
+        "party": ["enjoy", "indiferent", "hates"]
+    })
+    return df.group_by('party').agg(pl.col("age").mean())
```
Comment (matemq) [inline workshop/pipelines/exploration/nodes.py:10]: please sort the results
